test/e2e/node_reuse_test.go

```diff
@@ -455,15 +513,20 @@ func updateNodeReuse(nodeReuse bool, m3machineTemplateName string, clusterClient
 	Expect(m3machineTemplate.Spec.NodeReuse).To(BeTrue())
 }
 
-func untaintNodes(clientSet *kubernetes.Clientset, nodes *corev1.NodeList, taint *corev1.Taint) {
+func untaintNodes(clientSet *kubernetes.Clientset, nodes *corev1.NodeList, taint *corev1.Taint) (count int) {
+	count = 0
 	for i := range nodes.Items {
+		Logf("Untainting node %v ...", nodes.Items[i].Name)
```

Comment:
do we need this loop to untaint? Isn't it so that we can untaint all nodes at once using node labels?

Reply:
We can either wait by adding sleep for some time or loop until untainting the node. for some reason the taint is not present just after provisioning

Reply:
I am not convinced with this step. In previous step you assure that env has 3 updated CP bmh and machines are in running state. If CPs is running, it doesn't need untaint loop at all.

Reply:
@kashifest you mean `kubectl taint nodes --all node-role.kubernetes.io/master-` not sure if that feasible but I think at the end it end updating node by node.

@smoshiur1237 can you please refer to the step ? if you refer to the last cp untaint although all cp are running we we try to scale down the kcp to 1 if not all the nodes are untainted there are a probability that the nodes that will try still have the taint and the other nodes that are deprovisioning cannot migrate the pods to it

Reply:
> @kashifest you mean `kubectl taint nodes --all node-role.kubernetes.io/master-` not sure if that feasible but I think at the end it end updating node by node.
Yes and IMO thats what the command is suitable for, feasibility.

Reply:
I meant for the last untaint step.

Reply:
@smoshiur1237 Yes, if did not add that the last cp node that was upgraded will still have the taint. and the next step is scalling down kcp to 1 if that one was the node that has the untaint scaling down will hangs